### ISSUE TITLE
generate is broken on the latest commit

### ISSUE THREAD

**Issue by JackAKirk** [oneapi-src/unified-runtime#1625]
If I go to HEAD~100 it works but on tip:
I get:

```
~/unified-runtime$ cmake --build build --target generate
Traceback (most recent call last):
  File "run.py", line 13, in <module>
    import parse_specs
  File "/unified-runtime/scripts/parse_specs.py", line 553, in <module>
    def _make_versions(d, max_ver : Version) -> list[Version]:
TypeError: 'type' object is not subscriptable
make[7]: *** [CMakeFiles/generate-code.dir/build.make:70: CMakeFiles/generate-code] Error 1
make[6]: *** [CMakeFiles/Makefile2:1742: CMakeFiles/generate-code.dir/all] Error 2
make[5]: *** [CMakeFiles/Makefile2:1749: CMakeFiles/generate-code.dir/rule] Error 2
make[4]: *** [Makefile:582: generate-code] Error 2
make[3]: *** [CMakeFiles/generate.dir/build.make:70: CMakeFiles/generate] Error 2
make[2]: *** [CMakeFiles/Makefile2:1768: CMakeFiles/generate.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1775: CMakeFiles/generate.dir/rule] Error 2
make: *** [Makefile:595: generate] Error 2
```

**Comment by JackAKirk** [comment]
Seems to have been broken at some point between HEAD~10 and HEAD~25

**Comment by kbenzie** [comment]
This is not broken for me locally or in our GitHub Actions, I know the commit which introduced this change though. Its 577292e066127654a0cc105a9d03bd593c279b64 from https://github.com/oneapi-src/unified-runtime/pull/1582.

Could you share more info about your build environment, e.g. operating system, python version, are you running in a virtual environment?

**Comment by JackAKirk** [comment]
> This is not broken for me locally or in our GitHub Actions, I know the commit which introduced this change though. Its [577292e](https://github.com/oneapi-src/unified-runtime/commit/577292e066127654a0cc105a9d03bd593c279b64) from #1582.
> 
> Could you share more info about your build environment, e.g. operating system, python version, are you running in a virtual environment?

Python 3.8.10
ubuntu 20.04

I just followed the contributor instructions:
```
python3 -m venv .local
source .local/bin/activate
pip install -r third_party/requirements.txt
cmake -B build/ -DUR_FORMAT_CPP_STYLE=ON
cmake --build build --target generate
```

**Comment by kbenzie** [comment]
Okay, I suspect 20.04 might be where to look. I'll try reproducing in a container.

**Comment by kbenzie** [comment]
Should be resolved by #1626.

**Comment by JackAKirk** [comment]
> Should be resolved by #1626.

awesome, thanks
